eth_typing/abi.py

```diff
@@ -17,34 +17,17 @@
 """Binary data to be decoded."""
 
 
-class ABIEventComponent(TypedDict, total=False):
+class ABIParam(TypedDict, total=False):
     """
-    TypedDict to represent the `ABI` for nested event parameters.
-
-    Used as a component of `ABIEventParam`.
-    """
-
-    components: Sequence["ABIEventComponent"]
-    """List of nested event parameters for tuple event ABI types."""
-    name: str
-    """Name of the event parameter."""
-    type: str
-    """Type of the event parameter."""
-
-
-class ABIEventParam(TypedDict, total=False):
-    """
-    TypedDict to represent the `ABI` for event parameters.
+    TypedDict representing the `ABI` for error parameters.
```

Comment:
```suggestion
    TypedDict representing `ABI` components.
```

Reply:
Components here seems wrong to me, so maybe this is the case for using `ABIParam`.

The `ABIComponent` represents the inputs and outputs for functions/events, so maybe the doc should be something like:
```suggestion
    TypedDict representing the input and output parameters for an `ABI` interface.
```

Reply:
I can see that. To me, "param" feels more like a description of what it represents within the class itself, not really within the ABI specification.

The relevant part of the spec for this is [here](https://docs.soliditylang.org/en/latest/abi-spec.html#handling-tuple-types). This describes the ABI "tuple" type as having components but `total=False` means it can represent other descriptors as well. It really only labels it an ABI "object" which is kind of what it is - represented in JSON, we'd use the word object as well for something like `{"x": 1, "y": 2, "z": 3}`...

Do you still prefer `ABIParam` or:

- `ABIObject`: verbiage directly from that section
- `ABITypedVariable`: also taken directly from that section
- `ABIComponent`: "component" here taken as the definition of component... "a part or element of a larger whole" while also being used within `components`... or taken as "input" or "output" component objects

Thoughts there? I think I still lean component.

Reply:
Thanks for your reasoning, I do think using a more generic term to encompass the definition makes sense. For the `ABITypedVariable` it doesn't adapt as well when describing it as a nested object IMO. `ABIComponent` at least describes it at multiple levels, including the components that make up a Tuple.

Going with component.